sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -969,4 +969,9 @@ class JoinSuite extends QueryTest with SharedSQLContext {
           Seq(Row(0.0d, 0.0/0.0)))))
     }
   }
+
+  test("[SPARK-24783][SQL]spark.sql.shuffle.partitions=0 should throw exception ") {
+      val e = intercept[IllegalArgumentException](withSQLConf("spark.sql.shuffle.partitions" -> "0"){})
+      assert(e.getMessage.contains("the value of sql.shuffle.partitions must be greater than 0"))
+    }
```

Comment:
I think you can also move the test into https://github.com/apache/spark/blob/master/sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala